main/tests/server/src/com/google/refine/io/FileProjectManagerTests.java

```diff
@@ -92,4 +102,28 @@ public void saveReloadMultinationalCharacter() throws IOException {
         manager = new FileProjectManagerStub(workspaceDir);
         assertEquals(manager.getPreferenceStore().get("testPref"), "Refiné");
     }
+    
+    @Test
+    public void deleteProjectAndSaveWorkSpace() throws IOException {
+        FileProjectManager manager = new FileProjectManagerStub(workspaceFile);
+        /**
+         * String json = "{\n" + " \"preferences\" : {\n" + " \"entries\" : {\n" + " \"scripting.expressions\" : {\n" +
+         * " \"class\" : \"com.google.refine.preference.TopList\",\n" + " \"list\" : [ ],\n" + " \"top\" : 100\n" + "
+         * },\n" + " \"scripting.starred-expressions\" : {\n" + " \"class\" :
+         * \"com.google.refine.preference.TopList\",\n" + " \"list\" : [ ],\n" + " \"top\" : 2147483647\n" + " }\n" + "
+         * }\n" + " },\n" + " \"projectIDs\" : [ 5555 ]\n" + " }";
+         */
+
+        OutputStream stream = new FileOutputStream(workspaceFile);
+        InputStream f = new FileInputStream(workspaceFile);
+        ParsingUtilities.defaultWriter.writeValue(stream, json);
+        manager.saveWorkspace();
+        System.out.println(workspaceFile.getPath());
+        System.out.println(ParsingUtilities.inputStreamToString(f));
+
+        manager = new FileProjectManagerStub(workspaceFile);
+        manager.deleteProject(5555);
+        System.out.println(workspaceFile.getPath());
+        System.out.println(ParsingUtilities.inputStreamToString(f));
+    }
 }
```

Comment:
@elroykanye Can I get some assistance here. Seems am stuck.

Reply:
Hi @WaltonG , let me check this out and give you feedback.

Reply:
Similar to how you created a temporary workspace file, I think you should do same for the project file and call this `manager.saveToFile()` for it.

Am I on track with you?

Reply:
Sure. Let me look into that